lxd/storage/backend_lxd.go

```diff
@@ -672,8 +671,29 @@ func (b *lxdBackend) UnmountInstance(inst Instance, op *operations.Operation) (b
 	return b.driver.UnmountVolume(volType, volStorageName, op)
 }
 
+// GetInstanceDisk returns the type of disk for an instance and its location.
 func (b *lxdBackend) GetInstanceDisk(inst Instance) (string, string, error) {
-	return "", "", ErrNotImplemented
+	// tomp TODO get confirmation on "type" return value.
```

Comment:
@stgraber here is the implementation of GetInstanceDisk. Please can you explain further what the "type" return value should be? You mentioned "qcow" on the call the other day, but at this time I am converting the qcow to raw for VMs, as they will be effectively for other block devices. I'd like to understand what you intended the type value to be used for?

Reply:
For dir and btrfs, the on-disk storage is qcow2, for everything else, it's raw.
So on dir and btrfs, I'd expect it to return `("/var/lib/lxd/storage-pools/POOL/virtual-machines/NAME/root.img", "qcow2", nil)` and for all the others that support block, it should be `("/dev/...", "raw", nil)`.

Reply:
@stgraber right I see, and the returned format is then used in the qemue generated config to set the type, e.g.

```
format = "raw"
```

Reply:
@stgraber for dir/brtfs, do I just need to copy the image into the backend, rather than creating a new file and using qemu-convert?

Reply:
Right, for dir/btrfs, the filler should give you the expected path (not created), so you can just copy to that destination and then grow it to match the size (if one was passed). For the others, the destination path will exist as a block, in which case you know to qemu-img convert it to raw.